qhub/render/__init__.py

```diff
@@ -144,60 +144,35 @@ def render_template(output_directory, config_filename, force=False):
     )
 
 
-def remove_existing_renders(source_repo_dir, dest_repo_dir):
+def remove_existing_renders(dest_repo_dir, verbosity=0):
     """
-    Remove existing folder structure in output_dir apart from:
-    Files matching gitignore entries from the source template
-    Anything the user has added to a .qhubignore file in the output_dir (maybe their own github workflows)
-
-    No FILES in the dest_repo_dir are deleted.
-
-    The .git folder remains intact
+    Remove all files and directories beneath each directory in `deletable_dirs`. These files and directories will be regenerated in the next step (`generate_files`) based on the configurations set in `qhub-config.yml`.
 
     Inputs must be pathlib.Path
     """
-    copyfile(str(source_repo_dir / ".gitignore"), str(dest_repo_dir / ".gitignore"))
-
-    gitignore_matches = parse_gitignore(dest_repo_dir / ".gitignore")
-
-    if (dest_repo_dir / ".qhubignore").is_file():
-        qhubignore_matches = parse_gitignore(dest_repo_dir / ".qhubignore")
-    else:
+    home_dir = pathlib.Path.home()
+    if pathlib.Path.cwd() == home_dir:
+        raise ValueError(f"Deploying QHub from the home directory, {home_dir}, is not permitted.")
+
+    deletable_dirs = ["terraform-state", ".github", "infrastructure", "image"]
+
+    for deletable_dir in deletable_dirs:
+        deletable_dir = dest_repo_dir / deletable_dir
+        if deletable_dir.exists():
```

Comment:
Ah. Sorry I shouldn't have mentioned that `os.remove` and `shutil.rmtree` shouldn't be used. I meant that more generally that long term i'd like to minimize the number of times that we make this call! Since it's a scary operation.

I'm fine with using `shutil.rmtree` here and I think that will dramatically reduce the code here.T his should remove all the walking in the code.

Also `deletable_dirs` should include `.gitlab-ci.yml`.

Reply:
Got it! Thanks for the feedback. And I agree, using `shutil.rmtree` would be preferred.